x-pack/plugins/secops/public/containers/timeline/index.tsx

```diff
@@ -39,42 +46,41 @@ export class TimelineQuery extends QueryTemplate<
   GetTimelineQuery.Variables
 > {
   private updatedDate: number = Date.now();
-  private memoizedEvents: (events: EcsEdges[]) => Ecs[];
+  private memoizedTimelineEvents: (variables: string, events: TimelineEdges[]) => TimelineItem[];
 
   constructor(props: OwnProps) {
     super(props);
-    this.memoizedEvents = memoizeOne(this.getEcsEvent);
+    this.memoizedTimelineEvents = memoizeOne(this.getTimelineEvents);
   }
 
   public render() {
     const {
       children,
       id = 'timelineQuery',
       limit,
+      fields,
       filterQuery,
       poll,
       sourceId,
       sortField,
     } = this.props;
+    const variables = {
```

Comment:
You should be able to define your variables typescript like so
```suggest
const variables: GetTimelineQuery.Variables = {
```

Reply:
I thought that I was already protected by `<Query<GetTimelineQuery.Query, GetTimelineQuery.Variables>` because if you remove one attribute you will get an error

Reply:
This will give you protection from accidentally renaming an optional variable or adding additional variables that you do not want as it is more strict.